### PR TITLE
Created Disjoint Set Union in Cpp #4466

### PR DESCRIPTION
**Pull request by aryangulati**
**Fixes issue:**
Created Disjoint Set Union

**Changes:**

Added Disjoint Set Union

### CODE REVIEW

File: code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#include<bits/stdc++.h>
```
Comment (arnavb) [inline code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp]: #1915
Move code to its own folder
Don't use using namespace std

File: code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+using namespace std;
+
+#define int long long int
+#define ld long double
+#define F first
+#define S second
+#define pb push_back
```
Comment (arnavb) [inline code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp]: Don't use these defines except for the first one; use `lli` not `int`

File: code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const int N = 100001;
+
+int parent[N], sz[N];
+
+// gives superparent of the component
+int get_parent(int x) {
```
Comment (arnavb) [inline code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp]: camel case.

File: code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+void init() {
+    for (int i = 0; i < N; i++) {
+        parent[i] = i;
+        sz[i] = 1;
+    }
+}
+
+int32_t main()
```
Comment (arnavb) [inline code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp]: int

File: code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+{
+    init();
+    int n, m;
+    cin >> n >> m;
+    for (int i = 0; i < m; i++) {
```
Comment (arnavb) [inline code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp]: Pre increment.

File: code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp (hunk 2 above)
Comment (aryangulati) [reply code/graph_algorithms/src/Disjoint_Set_Union_algo.cpp]: basically for sake of using long long int i use int32_t instead it makes int as long long int so its easy and too applicable!

**Pull request closed without merging**